FreeRTOS_Sockets.c

```diff
@@ -1897,31 +1897,29 @@ BaseType_t FreeRTOS_setsockopt( Socket_t xSocket,
                     break;
             #endif /* ipconfigUSE_CALLBACKS */
 
-            #if ( ipconfigUSE_TCP != 0 )
-                #if ( ipconfigSOCKET_HAS_USER_SEMAPHORE != 0 )
+            #if ( ipconfigSOCKET_HAS_USER_WAKE_CALLBACK != 0 )
+                case FREERTOS_SO_WAKEUP_CALLBACK:
+
+                    /* Each socket can have a callback function that is executed
+                     * when there is an event the socket's owner might want to
+                     * process. */
+                    /* The type cast of the pointer expression "A" to type "B" removes const qualifier from the pointed to type. */
+                    pxSocket->pxUserWakeCallback = ( SocketWakeupCallback_t ) pvOptionValue;
+                    xReturn = 0;
+                    break;
+            #endif /* ipconfigSOCKET_HAS_USER_WAKE_CALLBACK */
+
+            #if ( ipconfigSOCKET_HAS_USER_SEMAPHORE != 0 )
+                case FREERTOS_SO_SET_SEMAPHORE:
 
                     /* Each socket has a semaphore on which the using task normally
                      * sleeps. */
-                    case FREERTOS_SO_SET_SEMAPHORE:
-                       {
-                           pxSocket->pxUserSemaphore = *( ipPOINTER_CAST( SemaphoreHandle_t *, pvOptionValue ) );
-                       }
-                        xReturn = 0;
-                        break;
-                #endif /* ipconfigSOCKET_HAS_USER_SEMAPHORE */
-
-                #if ( ipconfigSOCKET_HAS_USER_WAKE_CALLBACK != 0 )
```

Comment:
You took the 2 cases 'FREERTOS_SO_WAKEUP_CALLBACK' and 'FREERTOS_SO_SET_SEMAPHORE' out of an \#if block 'ipconfigUSE_TCP'. That is good.

But I don't know why you swapped the order of the two cases? Would you mind to swap them back as shown here?
~~~c
    case FREERTOS_SO_SET_SEMAPHORE:
    case FREERTOS_SO_WAKEUP_CALLBACK:
~~~

Thanks